### PR TITLE
feature(jenkins): Provides PowerShell scripts to start and stop the tutorials

### PR DESCRIPTION
**Pull request by gounthar**
One day, we'll do everything through a `docker-compose` file, but until that day, this set of scripts could help.

### CODE REVIEW

File: experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.ps1 [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+New-Item -ItemType Directory -Path $secretsDirectory -Force | Out-Null  # Creates the secrets directory if it doesn't exist
+
+$privateKeyPath = Join-Path $secretsDirectory "jenkins_agent_ed"  # Combines the secrets directory path with the private key filename
+$publicKeyPath = Join-Path $secretsDirectory "jenkins_agent_ed.pub"  # Combines the secrets directory path with the public key filename
+
+# Remove existing keys
+Remove-Item $privateKeyPath, $publicKeyPath -ErrorAction SilentlyContinue  # Removes any existing private and public key files
+
+# Generate new ed25519 SSH key pair
+ssh-keygen -t ed25519 -f $privateKeyPath -P "" -N ""  # Generates a new ed25519 SSH key pair with an empty passphrase
```
Comment (berviantoleo) [inline experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.ps1:18]: I think some considerations we need to add to the tutorial.

We should add the prerequisite. As far as I know, Windows user should install the ssh-keygen separately, which mean they should have this before running this script.

I have a suggestion to check all the prerequisites and show them which tools that should be installed before running the scripts.

Comment (gounthar) [reply experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.ps1:18]: Thank you @berviantoleo .
I happen to have `ssh-keygen` directly available within Powershell, but that's maybe it's a recent version.

Comment (berviantoleo) [reply experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.ps1:18]: Ah, okay. I just found [an article](https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_keymanagement#user-key-generation).

We can assume the ssh-keygen is already there.

File: jenkins_init.ps1 [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    param(
+        [string]$tutorialPath
+    )
+
+    Write-Host "Starting tutorial $tutorialPath"
+    & $DOCKER_COMPOSE -f "$tutorialPath/docker-compose.yaml" up -d
+}
+
+# Check if tutorials are already running
+Check-RunningTutorials
```
Comment (berviantoleo) [inline jenkins_init.ps1:53]: Compared with the bash one.

https://github.com/ash-sxn/GSoC-2023-docker-based-quickstart/blob/main/jenkins_init.sh#L62

I think we also need to check if docker and docker-compose exist.

**Pull request closed without merging**